tests/h/util/group_scope_test.py

```diff
@@ -28,6 +28,50 @@ def test_it_parses_scope_from_uri(uri, expected_scope):
     assert scope == expected_scope
 
 
+class TestURIInScope(object):
+    @pytest.mark.parametrize(
+        "uri,in_origin,in_path,in_other",
+        [
+            ("https://www.foo.com", True, False, False),
+            ("http://foo.com", False, False, False),
+            ("http://www.foo.com/bar/qux.html", True, True, True),
+            ("http://www.foo.com/bar/baz/qux.html", True, True, True),
+            ("http://foo.com/", False, False, False),
+            ("http://www.foo.com/bar.baz", True, False, False),
+            ("www.foo.com", False, False, False),
+            ("randoscheme://foo.com", False, False, False),
+            ("foo", False, False, False),
+            ("https://www.foo.com/bar/baz", True, False, False),
+            ("", False, False, False),
+        ],
+    )
+    def test_it_returns_True_if_uri_matches_one_or_more_scopes(
```

Comment:
I find it tricky to see that this test covers the relevant cases. What I think I would have found more helpful is to have separate tests for checking:

- The URI matching aspect of the behavior (set of tests checking one URI against one scope)
-  The "at least one scope must match" aspect of the behavior